sql/src/main/java/io/crate/executor/transport/SnapshotRestoreDDLDispatcher.java

```diff
@@ -169,77 +173,96 @@ public void onFailure(Exception e) {
     }
 
     @VisibleForTesting
-    static CompletableFuture<List<String>> resolveIndexNames(List<RestoreSnapshotAnalyzedStatement.RestoreTableInfo> restoreTables, boolean ignoreUnavailable,
-                                                             TransportGetSnapshotsAction getSnapshotsAction, String repositoryName) {
-        Collection<String> resolvedIndices = new HashSet<>();
+    static CompletableFuture<ResolveIndicesAndTemplatesContext> resolveIndexNames(List<RestoreSnapshotAnalyzedStatement.RestoreTableInfo> restoreTables, boolean ignoreUnavailable,
+                                                                                  TransportGetSnapshotsAction getSnapshotsAction, String repositoryName) {
+        ResolveIndicesAndTemplatesContext resolveIndicesAndTemplatesContext = new ResolveIndicesAndTemplatesContext();
         List<RestoreSnapshotAnalyzedStatement.RestoreTableInfo> toResolveFromSnapshot = new ArrayList<>();
         for (RestoreSnapshotAnalyzedStatement.RestoreTableInfo table : restoreTables) {
             if (table.hasPartitionInfo()) {
-                resolvedIndices.add(table.partitionName().asIndexName());
+                resolveIndicesAndTemplatesContext.addIndex(table.partitionName().asIndexName());
+                resolveIndicesAndTemplatesContext.addTemplate(PartitionName.templateName(table.tableIdent().schema(), table.tableIdent().name()));
```

Comment:
think it would make sense to add the templateName to RestoreTableInfo to avoid repeating..